src/Compiler/Interpreter/VMCode.idr

```diff
@@ -0,0 +1,281 @@
+        res <- callPrim stk fn argObjs
+        setReg stk target res
+    step stk (EXTPRIM target fn args) = case lookup fn knownExtern of
+        Nothing => interpError stk $ "EXTPRIM: Unkown foreign function: " ++ show fn
+        Just (ar ** op) => case toVect ar args of
+            Nothing => interpError stk $ "EXTPRIM: Wrong number of arguments, found: " ++ show (length args) ++ ", expected: " ++ show ar
+            Just argsVect => do
+                argObjs <- traverseVect (getReg stk) argsVect
+                res <- op stk argObjs
+                setReg stk target res
+    step stk (CASE sc alts def) = do
+        scObj <- getReg stk sc
+        case scObj of
+            Constructor tag _ => matchCon stk tag alts def
+            _ => interpError stk $ "CASE: Expected Constructor, found " ++ showType scObj
+      where
+        matchCon : Stack -> Either Int Name -> List (Either Int Name, List VMInst) -> Maybe (List VMInst) -> Core ()
+        matchCon stk tag [] Nothing = interpError stk "CASE: Missing matching alternative or default"
+        matchCon stk tag [] (Just is) = traverse_ (step stk) is
+        matchCon stk tag ((t, is) :: alts) def =
+            if tag == t
+                then traverse_ (step stk) is
+                else matchCon stk tag alts def
+    step stk (CONSTCASE sc alts def) = do
+        scObj <- getReg stk sc
+        case scObj of
+            Const c => matchConst stk c alts def
+            _ => interpError stk $ "CONSTCASE: Expected Constant, found " ++ showType scObj
+      where
+        matchConst : Stack -> Constant -> List (Constant, List VMInst) -> Maybe (List VMInst) -> Core ()
+        matchConst stk c [] Nothing = interpError stk "CONSTCASE: Missing matching alternative or default"
+        matchConst stk c [] (Just is) = traverse_ (step stk) is
+        matchConst stk c ((c', is) :: alts) def =
+            if c == c'
+                then traverse_ (step stk) is
+                else matchConst stk c alts def
+    step stk (PROJECT target val idx) = do
+        valObj <- getReg stk val
+        case valObj of
+            Constructor _ args => case indexMaybe args idx of
+                Nothing => interpError stk
+                    $ "PROJECT: Unable to project index " ++ show idx
+                    ++ ", missing arguments for constructor:\n" ++ show valObj
+                Just arg => setReg stk target arg
+            _ => interpError stk $ "PROJECT: Expected Constructor, found " ++ showType valObj
+    step stk (NULL reg) = setReg stk reg Null
+    step stk (ERROR msg) = interpError stk $ "ERROR: " ++ msg
+
+    callFunc : Ref State InterpState => Stack -> Name -> List Object -> Core Object
+    callFunc stk fn args = saveLocals $ do
+        let ind = pack $ '|' <$ stk
+        logCallStack <- logging "compiler.interpreter" 25
+        when logCallStack $ coreLift $ putStrLn $ ind ++ "Calling " ++ show fn ++ " with args: " ++ show args
```

Comment:
AFAIU this is just

```suggestion
        log "compiler.interpreter" 25 $ ind ++ "Calling " ++ show fn ++ " with args: " ++ show args
```

Reply:
I think `log` would add more noise at the start of the line.